frontend/public/components/cluster-settings/cluster-settings.tsx

```diff
@@ -69,6 +88,54 @@ const cancelUpdate = (cv: ClusterVersionKind) => {
 
 export const clusterAutoscalerReference = referenceForModel(ClusterAutoscalerModel);
 
+const getMachineConfigPoolByName = (
+  machineConfigPools: MachineConfigPoolKind[],
+  name: string,
+): MachineConfigPoolKind => {
+  return machineConfigPools?.find((mcp) => mcp.metadata.name === name);
+};
+
+const getStartedTimeForCVDesiredVersion = (
+  cv: ClusterVersionKind,
+  desiredVersion: string,
+): string => {
+  const desiredHistory: UpdateHistory = cv?.status?.history?.find(
+    (update) => update.version === desiredVersion,
+  );
+  return desiredHistory?.startedTime;
+};
+
+const getUpdatingTimeForMCP = (machineConfigPool: MachineConfigPoolKind): string => {
+  const updatingCondition = machineConfigPool?.status?.conditions.find(
+    (condition) => condition.type === MachineConfigPoolConditionType.Updating,
+  );
+  return updatingCondition?.lastTransitionTime;
+};
+
+const getUpdatedOperatorCount = (
+  clusterOperators: ClusterOperator[],
+  desiredVersion: string,
+): number => {
+  return clusterOperators?.filter((operator) => {
+    const progressingCondition = operator?.status?.conditions.find(
+      (condition) => condition.type === 'Progressing',
+    );
+    return (
+      getClusterOperatorVersion(operator) === desiredVersion &&
+      progressingCondition?.status === 'False'
+    );
+  })?.length;
```

Comment:
I'd return `0` instead of `undefined` when there are no cluster operators. (That avoids needing to think about a possible `undefined > 0` check in `getIsUpdating`.)

```suggestion
  })?.length ?? 0;
```